### PR TITLE
A_mod suppression of non-linear power

### PR DESCRIPTION
**Pull request by ajouellette**
This is a very simple module to sample the phenomenological parameter A_mod from Amon & Efstathiou 2022.

### CODE REVIEW

File: structure/amod/module.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: amod
+version: 1.0
+purpose: Modify the non-linear matter power spectrum using the A_mod phenomenological parameterization
+url: ""
+interface: amod.py
+attribution: []
```
Comment (joezuntz) [inline structure/amod/module.yaml:6]: Thanks so much for this contribution! You should definitely give yourself credit by adding your name to the attribution list!

File: structure/amod/module.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    non-linear power spectrum.
+
+    :math:`P_m(k, z) = P_m^L(k, z) + A_{mod} [P_m^{NL}(k, z) - P_m^L(k, z)]`
+
+    To match the model used in Amon & Efstathiou (2022), the non-linear power spectrum should
+    be computed using HMCode2020 without baryonic feedback.
+    "
+
+# List of parameters that can go in the params.ini file in the section for this module
+params:
```
Comment (joezuntz) [inline structure/amod/module.yaml:26]: This block should be replaced with:

```
params: {}
```

File: structure/amod/amod.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def execute(block):
+    # inputs
+    pk_lin = block["matter_power_lin", "p_k"]
+    pk_nonlin = block["matter_power_nl", "p_k"]
+    A_mod = block["amod_parameter", "a_mod"]
+
+    # for now assume pk_lin and pk_nonlin are defined on the same grids
+    assert (block["matter_power_lin", "z"] == block["matter_power_nl", "z"]).all()
```
Comment (joezuntz) [inline structure/amod/amod.py:15]: Could you replace these with this, just to give some more info to the user:
```
z_same = np.allclose(block["matter_power_lin", "z"], block["matter_power_nl", "z"])
k_same = np.allclose(block["matter_power_lin", "k_h"], block["matter_power_nl", "k_h"])
if not k_same and z_same:
    raise ValueError("The redshift and k sampling of the linear and non-linear matter power must be the same to use the amod module")

**Review by joezuntz — CHANGES_REQUESTED** [review]
Thanks so much for this! I have three pretty minor requests for changes and then I'd love to merge it.

**Comment by ajouellette** [comment]
Cool! I just added the suggested changes.

**Review by joezuntz — APPROVED** [review]

**Comment by joezuntz** [comment]
Thanks so much!

**Pull request merged**